### PR TITLE
Adds a manual crank cell charger

### PR DESCRIPTION
**Pull request by JustSumBody**
Revival of #4947
Constructable from an APC frame, metal rod, cables, and wirecutters
Insert a power cell and attack_self it to crank it for a whopping TEN ~WATTS~ JOULES per crank

Video: https://dl.dropboxusercontent.com/s/md3dytttf6o2iu6/2017-01-21_07-10-31.mp4

~~NOTE THAT FOR SOME REASON CRAFTING ONE MAKES TWO THIS IS NOT INTENDED AND I'M NOT SURE HOW TO FIX~~
No longer makes the result twice, but it throws the same debug warning that wall mounted holomaps throw when you finish the last step:
```ADMIN LOG: Holy shit /datum/construction/reversible/crank_charger/(/datum/construction/reversible/crank_charger) is trying to set its new index to 0! how the fuck did this happen? I don't know, our direction is forward old index was 1. User is justsumguy, itemused the screwdriver, step /list```
I'm not sure if what's going on is a result of construction datum code itself or my/Deity's use of it. @ComicIronic 

Besides that, I guess it could do with construction sprites or an inhand but eh.

### CODE REVIEW

**Comment by yclatious** [comment]
Even more ghetto tools, nice.

**Comment by PJB3005** [comment]
> watts per crank.

Basic physics class is that way ➡️

File: code/game/machinery/cell_charger.dm [hunk 1]
```diff
@@ -144,3 +144,104 @@
...
+/obj/item/device/crank_charger/examine(mob/user)
+	..()
+	if(stored)
+		to_chat(user,"<span class='info'>The readout displays: [round(stored.charge/stored.maxcharge*100)]%.</span>")
+	else
+		to_chat(user,"<span class='info'>There is no cell loaded.</span>")
+
+/obj/item/device/crank_charger/attackby(obj/item/W, mob/user)
+	if(!stored && istype(W,/obj/item/weapon/cell))
+		stored = W
```
Comment (PJB3005) [inline code/game/machinery/cell_charger.dm:227]: This should check whether it's even possible to drop the item before actually storing it.

File: code/game/machinery/cell_charger.dm [hunk 2]
```diff
@@ -144,3 +144,104 @@
...
+/obj/item/device/crank_charger/attack_hand(mob/user)
+	if(stored && user.get_inactive_hand() == src)
+		stored.loc = user.loc
+		user.put_in_hands(stored)
+		stored = null
+		update_icon()
+	else
+		..()
+		
+/obj/item/device/crank_charger/Destroy()
```
Comment (PJB3005) [inline code/game/machinery/cell_charger.dm:253]: This needs to do a super call.

File: code/game/machinery/cell_charger.dm [hunk 3]
```diff
@@ -144,3 +144,104 @@
...
+			update_icon()
+			playsound(get_turf(src), 'sound/items/crank.ogg',50,1)
+			if(stored.charge>stored.maxcharge)
+				stored.charge = stored.maxcharge
+	else
+		to_chat(user,"<span class='warning'>There is no cell loaded!</span>")
+
+/obj/item/device/crank_charger/attack_hand(mob/user)
+	if(stored && user.get_inactive_hand() == src)
+		stored.loc = user.loc
```
Comment (PJB3005) [inline code/game/machinery/cell_charger.dm]: Put on hands should already handle this.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by SkowronX** [comment]
This isnt how energy workssss REEEEEEEEEEE

**Comment by BobdaBiscuit** [comment]
Would it be possible to have the lights on the frame indicate how charged the cell is, if it doesn't already?

**Comment by Kurfursten** [comment]
@BobdaBiscuit It would be possible but only if someone sprited it

**Comment by Shadowmech88** [comment]
I think you need to add some sort of miniscule delay to the cranking, else people with autoclickers will be churning out a ton of power due to there being no delay on attack_self() by default.

**Comment by PJB3005** [comment]
There's a delay of one decisecond.

**Comment by kol1th** [comment]
People use autoclickers?

**Comment by JustSumBody** [comment]
>he doesn't use an autoclicker

**Comment by Intigracy** [comment]
My mouse came with a button that automatically clicks 3x.

Welcome to the future.

File: code/game/machinery/cell_charger.dm [hunk 4]
```diff
@@ -212,7 +212,7 @@
 		to_chat(user,"<span class='info'>There is no cell loaded.</span>")
 
 /obj/item/device/crank_charger/attackby(obj/item/W, mob/user)
-	if(!stored && istype(W,/obj/item/weapon/cell))
+	if(!W.cant_drop && !stored && istype(W,/obj/item/weapon/cell))
 		stored = W
 		user.drop_item(W,src)
```
Comment (PJB3005) [inline code/game/machinery/cell_charger.dm]: Assuming that can't drop js the only thing that can make an item not be droppable is a very wild assumption.

**Comment by Exxion** [comment]
Now that vaults exist and thus space exploration is actually relevant, this will be much more useful for charging spacepods on the go

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/machinery/cell_charger.dm (hunk 4 above)
Comment (ComicIronic) [reply code/game/machinery/cell_charger.dm]: It's also not encapsulated. You should just check the result of `drop_item`.

**Comment by gbasood** [comment]
quality Metro 2033 gameplay

**Comment by Kitfox88** [comment]
jack in the boxes when

**Review by PJB3005 — APPROVED** [review]

File: code/game/machinery/cell_charger.dm (hunk 2 above)
Comment (Probe1) [reply code/game/machinery/cell_charger.dm:253]: @JustSumBody

**Comment by SonixApache** [comment]
~~UPDATE WIKI~~
I UPDATED THE WIKI FOR YOU

**Pull request merged**